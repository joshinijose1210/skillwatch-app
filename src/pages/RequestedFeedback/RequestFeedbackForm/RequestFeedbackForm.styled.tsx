import bulbOn from '@constants/images/icons/lightbulb_off.png';
import bulbOff from '@constants/images/icons/lightbulb_on.png';
import { Text } from '@medly-components/core';
import styled, { css } from 'styled-components';
import { MultiSelect } from '@medly-components/core';
import { StyledToggleWrapper } from '@common';

const getBackgroundColor = (theme: any, feedbackTag?: number) => {
    return css`
        background-color: ${theme.colors.grey[50]};
    `;
};

export const FlexBoxRow = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 4rem;
`;

export const SendButtonBox = styled.div`
    display: flex;
    justify-content: end;
    gap: 1rem;
`;

export const FlexBoxColumn = styled.div`
    display: flex;
    flex-direction: column;
    row-gap: 1rem;
    width: 100%;
    min-width: 500px;
`;

export const ErrorBox = styled.div`
    display: flex;
    justify-content: end;
`;

export const StyledContainer = styled.div`
    display: flex;
`;

export const BulbImg = styled.div`
    background-image: url(${bulbOff});
    background-position: center;
    background-size: contain;
    width: 2.8rem;
    height: 2.8rem;
    background-repeat: no-repeat;
    margin-left: 0.5rem;
`;

export const TipContainer = styled.div<{ feedbackTag?: number }>`
    ${({ feedbackTag, theme }) => getBackgroundColor(theme, feedbackTag)}
    margin: 0 0 2rem 2rem;
    border-radius: 1rem;
    padding: 2rem;
    color: ${({ theme }) => theme.colors.grey[600]};
    &:hover ${BulbImg} {
        background-image: url(${bulbOn}) !important;
    }
`;

export const TipHeaderText = styled.div`
    color: black;
    display: flex;
    align-items: flex-end;
    margin-bottom: 2rem;

    ${Text.Style} {
        line-height: unset;
    }
`;

export const ToggleWrapper = styled(StyledToggleWrapper)`
    width: calc(50% - 20px);

    div {
        display: flex;
        justify-content: space-between;
    }
`;

export const MultiSelectWrapper = styled(MultiSelect)`
    max-height: 60px;
    max-width: 25rem;
    margin: 0;

    #enteremail-selected-chips {
        height: fit-content;
    }

    div > div {
        height: 4.5rem;
    }

    div > div > input {
        line-height: 1.9rem;
        font-size: 1.3rem;
        &::placeholder {
            color: transparent;
        }
    }

    div > div > label {
        line-height: 2.1rem;
        color: #666 !important;
    }
    > :last-child {
        min-width: 100%;
        width: fit-content;
        ${Text.Style} {
            font-size: 1.3rem;
        }
        > ul {
            max-height: 30rem;
        }
    }
`;
