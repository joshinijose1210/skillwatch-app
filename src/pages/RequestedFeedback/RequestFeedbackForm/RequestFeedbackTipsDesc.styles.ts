import { Text } from '@medly-components/core';
import styled from 'styled-components';

export const TipHeader = styled(Text)`
    margin: 1rem 0;
`;

export const RequestFeedbackTipHeader = styled(Text)<{ margin?: string }>`
    margin: ${({ margin }) => `${margin}`};
`;

export const StepsList = styled.ul`
    list-style: none;
`;

export const Step = styled.li`
    margin-top: 1rem;
    position: relative;
    &::before {
        display: inline-block;
        content: '';
        border-radius: 0.375rem;
        height: 0.75rem;
        width: 0.75rem;
        margin-right: 0.5rem;
        position: absolute;
        left: -35px;
        top: 5px;
        background-color: #bdbdbd;
    }
    ::marker {
        top: 0;
    }
`;
