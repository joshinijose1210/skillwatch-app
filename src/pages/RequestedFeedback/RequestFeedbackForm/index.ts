export { RequestFeedbackForm as default } from './RequestFeedbackForm';
