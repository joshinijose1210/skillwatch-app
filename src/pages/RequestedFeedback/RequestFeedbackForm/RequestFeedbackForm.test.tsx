import { RTKRender, cleanup, fireEvent, screen, waitFor } from '@utils/test-utils';
import { RequestFeedbackForm } from './RequestFeedbackForm';
import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import { ManagerReviewMockData } from '@mocks/managerReview';
import * as reactRouterDom from 'react-router-dom';
import { feedbackFormMockData } from '@mocks/feedbackForm';
import { userMockState } from '@mocks/userMockState';

const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
    useLocation: jest.fn()
}));

beforeEach(() => {
    window.HTMLElement.prototype.scrollIntoView = jest.fn();
    const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');
    useLocationMock.mockReturnValue({
        ...ManagerReviewMockData.LocationMockData,
        state: { ...feedbackFormMockData.mockState, action: 'Add' }
    });
    const test = 'Request feedback form component should render with proper data if view any request';
    if (expect.getState().currentTestName?.includes(test)) {
        useLocationMock.mockReturnValue({
            ...ManagerReviewMockData.LocationMockData,
            pathname: 'my-feedback/Request-Feedback/1/View-Requested-Feedback',
            state: {
                ...feedbackFormMockData.mockState,
                action: 'View',
                isExternalRequest: true,
                externalFeedbackFromEmail: 'test@gmail.com'
            }
        });
    }
    cleanup();
});

afterEach(cleanup);

describe('Request feedback form component', () => {
    it('should take snapshot', () => {
        const { container } = RTKRender(<RequestFeedbackForm />);
        expect(container).toMatchSnapshot();
    });

    it('should render propely', () => {
        RTKRender(<RequestFeedbackForm />);
        expect(screen.getByText('Request')).toBeInTheDocument();
        const SentBtn = screen.getByTestId('sendBtn');
        expect(SentBtn).toBeInTheDocument();
        expect(SentBtn).toBeDisabled();
    });

    it('should add request if correct data is entered', async () => {
        const { url, data2 } = feedbackFormMockData.employee;
        mock.onGet(url).replyOnce(200, data2);
        const { url: actionItemUrl, data: actionItemData } = feedbackFormMockData.actionItems;
        mock.onGet(actionItemUrl).replyOnce(200, actionItemData);
        mock.onPost('https://dummy-api.example.com/api/feedback-request/').replyOnce(200, {});
        const { container, rerender } = RTKRender(<RequestFeedbackForm />);
        rerender(<RequestFeedbackForm />);
        const input = container.querySelector('#feedbackfrom-input');
        expect(input).not.toBeNull();
        fireEvent.click(screen.getByTestId('employeeOne'));
        await screen.findByText('T S (SR0007)');
        fireEvent.click(screen.getByText('T S (SR0007)'));
        await new Promise(resolve => setTimeout(resolve, 0));
        fireEvent.click(screen.getByTestId('employeeTwo'));
        await screen.findByText('A S (123)');
        fireEvent.click(screen.getByText('A S (123)'));
        await new Promise(resolve => setTimeout(resolve, 0));
        const editor = container
            .querySelector('[data-placeholder="Please mention the project/task for which you want the feedback"]')
            ?.getElementsByTagName('p');
        if (editor !== undefined) {
            fireEvent.change(editor[0], {
                target: { innerHTML: 'This a test feedback request just to verify the functionality of the component.' }
            });
        }
        const SentBtn = screen.getByTestId('sendBtn');
        await waitFor(() => expect(SentBtn).toBeEnabled(), { timeout: 3000 });
        fireEvent.click(SentBtn);
        await screen.findByText('Request Feedback submitted successfully');
    });

    it('should check if external email input and validations are working correctly', async () => {
        const { url, data2 } = feedbackFormMockData.employee;
        mock.onGet(url).replyOnce(200, data2);
        const { url: actionItemUrl, data: actionItemData } = feedbackFormMockData.actionItems;
        mock.onGet(actionItemUrl).replyOnce(404, actionItemData);
        const { container } = RTKRender(<RequestFeedbackForm />, { initialState: userMockState });
        expect(screen.getByText('Request')).toBeInTheDocument();
        const SentBtn = screen.getByTestId('sendBtn');
        fireEvent.click(screen.getByTestId('toggle'));
        fireEvent.click(screen.getByLabelText('Enter email'));
        fireEvent.change(screen.getByLabelText('Enter email'), { target: { value: 'external@gmail.com' } });
        fireEvent.click(screen.getByText('Create "external@gmail.com"'));
        await waitFor(() => expect(screen.getByLabelText('Enter email')).toHaveValue(''));
        fireEvent.change(screen.getByLabelText('Enter email'), { target: { value: 'abc@scalereal.com' } });
        fireEvent.click(screen.getByText('Create "abc@scalereal.com"'));
        expect(await screen.findByText('Cannot send request to same user.')).toBeInTheDocument();
        const editor = container
            .querySelector('[data-placeholder="Please mention the project/task for which you want the feedback"]')
            ?.getElementsByTagName('p');
        if (editor !== undefined) {
            fireEvent.change(editor[0], { target: { innerHTML: 'limit should be of 50 characters' } });
        }
        await screen.findByText('Please write more than 50 characters.');
        expect(SentBtn).toBeDisabled();
    });

    it('should render with proper data if view any request', () => {
        const { url, data } = feedbackFormMockData.employee;
        mock.onGet(url).replyOnce(200, data);
        RTKRender(<RequestFeedbackForm />);
        expect(screen.getByText('View Requested Feedback')).toBeInTheDocument();
        expect(screen.getByText('Test request')).toBeInTheDocument();
        expect(screen.getByText('requested data')).toBeInTheDocument();
    });
});
