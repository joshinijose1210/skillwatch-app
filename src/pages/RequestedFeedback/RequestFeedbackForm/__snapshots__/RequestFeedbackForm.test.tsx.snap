// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request feedback form component should take snapshot 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Request Feedback
        </h3>
      </div>
    </div>
    <div
      class="sc-tkiih6-29 sc-uyelpa-8 fWtWBE lIoyJ"
    >
      <div
        class="sc-12atfcu-3 eAlGLr"
        id="request-for-feedback-from-a-client/stakeholder-?"
        labelposition="left"
      >
        <label
          class="sc-975cgx-0 aFzmi sc-12atfcu-1 gBASdL"
          for="request-for-feedback-from-a-client/stakeholder-?"
          labelposition="left"
          labelvariant="h4"
          labelweight="Regular"
        >
          Request for feedback from a client/stakeholder ?
        </label>
        <div
          class="sc-12atfcu-0 sc-1oxfop0-0 hjFLkE hzRUNj"
        >
          <input
            class="sc-1oxfop0-5 kOEioh sc-tkiih6-30"
            data-testid="toggle"
            id="request-for-feedback-from-a-client/stakeholder-?"
            type="checkbox"
          />
          <div
            class="sc-1oxfop0-1 kTppyc"
          >
            <div
              class="sc-1oxfop0-4 fOqcXx"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-uyelpa-4 jNXVNA"
    >
      <div
        action="Add"
        class="sc-tkiih6-9 bFDgcw form-division"
      >
        <div
          class="sc-11tbpdo-0 fSZVnk sc-tkiih6-14 gSveWN"
          data-testid="employeeOne"
          id="feedbackfrom-wrapper"
          minwidth="100%"
          variant="outlined"
        >
          <div
            class="sc-17d4s39-0 khebHA"
            id="feedbackfrom-input-wrapper"
            minwidth="100%"
          >
            <div
              class="sc-19px6zp-0 BrSGZ"
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="feedbackfrom-helper-text"
                  autocomplete="off"
                  class="sc-hb99wz-0 jaEbhp"
                  errortext=""
                  id="feedbackfrom-input"
                  inputsize="M"
                  placeholder="Select Employee Name"
                  type="text"
                  value=""
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="feedbackfrom-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Feedback From
                </label>
              </div>
              <div
                class="sc-4vbg19-0 hFRmRP"
                id="feedbackfrom-count"
              >
                <svg
                  class="sc-1pf3x8a-0 klDMsE"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                    fill="#000"
                  />
                </svg>
              </div>
            </div>
            
          </div>
        </div>
        
        <div
          class="sc-11tbpdo-0 fSZVnk sc-tkiih6-14 gSveWN"
          data-testid="employeeTwo"
          id="feedbackabout-wrapper"
          minwidth="100%"
          variant="outlined"
        >
          <div
            class="sc-17d4s39-0 khebHA"
            id="feedbackabout-input-wrapper"
            minwidth="100%"
          >
            <div
              class="sc-19px6zp-0 BrSGZ"
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="feedbackabout-helper-text"
                  autocomplete="off"
                  class="sc-hb99wz-0 jaEbhp"
                  errortext=""
                  id="feedbackabout-input"
                  inputsize="M"
                  placeholder="Select Employee Name"
                  type="text"
                  value=""
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="feedbackabout-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Feedback About
                </label>
              </div>
              <div
                class="sc-4vbg19-0 hFRmRP"
                id="feedbackabout-count"
              >
                <svg
                  class="sc-1pf3x8a-0 klDMsE"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                    fill="#000"
                  />
                </svg>
              </div>
            </div>
            
          </div>
        </div>
        
        <div
          class="sc-1wh81lg-1 cwMleO"
        >
          <div
            class="sc-1wh81lg-0 RMuwJ"
            editor="feedback"
          >
            <div
              class="ql-toolbar ql-snow"
            >
              <span
                class="ql-formats"
              >
                <button
                  class="ql-bold"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <path
                      class="ql-stroke"
                      d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
                    />
                     
                    <path
                      class="ql-stroke"
                      d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-italic"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="13"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="5"
                      x2="11"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="8"
                      x2="10"
                      y1="14"
                      y2="4"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-underline"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <path
                      class="ql-stroke"
                      d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
                    />
                     
                    <rect
                      class="ql-fill"
                      height="1"
                      rx="0.5"
                      ry="0.5"
                      width="12"
                      x="3"
                      y="15"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-strike"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke ql-thin"
                      x1="15.5"
                      x2="2.5"
                      y1="8.5"
                      y2="9.5"
                    />
                     
                    <path
                      class="ql-fill"
                      d="M9.007,8C6.542,7.791,6,7.519,6,6.5,6,5.792,7.283,5,9,5c1.571,0,2.765.679,2.969,1.309a1,1,0,0,0,1.9-.617C13.356,4.106,11.354,3,9,3,6.2,3,4,4.538,4,6.5a3.2,3.2,0,0,0,.5,1.843Z"
                    />
                     
                    <path
                      class="ql-fill"
                      d="M8.984,10C11.457,10.208,12,10.479,12,11.5c0,0.708-1.283,1.5-3,1.5-1.571,0-2.765-.679-2.969-1.309a1,1,0,1,0-1.9.617C4.644,13.894,6.646,15,9,15c2.8,0,5-1.538,5-3.5a3.2,3.2,0,0,0-.5-1.843Z"
                    />
                     
                  </svg>
                </button>
              </span>
              <span
                class="ql-formats"
              >
                <span
                  class="ql-align ql-picker ql-icon-picker"
                >
                  <span
                    aria-controls="ql-picker-options-0"
                    aria-expanded="false"
                    class="ql-picker-label"
                    role="button"
                    tabindex="0"
                  >
                    <svg
                      viewBox="0 0 18 18"
                    >
                       
                      <line
                        class="ql-stroke"
                        x1="3"
                        x2="15"
                        y1="9"
                        y2="9"
                      />
                       
                      <line
                        class="ql-stroke"
                        x1="3"
                        x2="13"
                        y1="14"
                        y2="14"
                      />
                       
                      <line
                        class="ql-stroke"
                        x1="3"
                        x2="9"
                        y1="4"
                        y2="4"
                      />
                       
                    </svg>
                  </span>
                  <span
                    aria-hidden="true"
                    class="ql-picker-options"
                    id="ql-picker-options-0"
                    tabindex="-1"
                  >
                    <span
                      class="ql-picker-item ql-selected"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        viewBox="0 0 18 18"
                      >
                         
                        <line
                          class="ql-stroke"
                          x1="3"
                          x2="15"
                          y1="9"
                          y2="9"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="3"
                          x2="13"
                          y1="14"
                          y2="14"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="3"
                          x2="9"
                          y1="4"
                          y2="4"
                        />
                         
                      </svg>
                    </span>
                    <span
                      class="ql-picker-item"
                      data-value="center"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        viewBox="0 0 18 18"
                      >
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="9"
                          y2="9"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="14"
                          x2="4"
                          y1="14"
                          y2="14"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="12"
                          x2="6"
                          y1="4"
                          y2="4"
                        />
                         
                      </svg>
                    </span>
                    <span
                      class="ql-picker-item"
                      data-value="right"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        viewBox="0 0 18 18"
                      >
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="9"
                          y2="9"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="5"
                          y1="14"
                          y2="14"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="9"
                          y1="4"
                          y2="4"
                        />
                         
                      </svg>
                    </span>
                    <span
                      class="ql-picker-item"
                      data-value="justify"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        viewBox="0 0 18 18"
                      >
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="9"
                          y2="9"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="14"
                          y2="14"
                        />
                         
                        <line
                          class="ql-stroke"
                          x1="15"
                          x2="3"
                          y1="4"
                          y2="4"
                        />
                         
                      </svg>
                    </span>
                  </span>
                </span>
                <select
                  class="ql-align"
                  style="display: none;"
                >
                  <option
                    selected="selected"
                  />
                  <option
                    value="center"
                  />
                  <option
                    value="right"
                  />
                  <option
                    value="justify"
                  />
                </select>
              </span>
              <span
                class="ql-formats"
              >
                <button
                  class="ql-list"
                  type="button"
                  value="ordered"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke ql-thin"
                      x1="2.5"
                      x2="4.5"
                      y1="5.5"
                      y2="5.5"
                    />
                     
                    <path
                      class="ql-fill"
                      d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
                    />
                     
                    <path
                      class="ql-stroke ql-thin"
                      d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
                    />
                     
                    <path
                      class="ql-stroke ql-thin"
                      d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-list"
                  type="button"
                  value="bullet"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="14"
                      y2="14"
                    />
                     
                  </svg>
                </button>
              </span>
              <span
                class="ql-formats"
              >
                <button
                  class="ql-link"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="11"
                      y1="7"
                      y2="11"
                    />
                     
                    <path
                      class="ql-even ql-stroke"
                      d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
                    />
                     
                    <path
                      class="ql-even ql-stroke"
                      d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
                    />
                     
                  </svg>
                </button>
              </span>
            </div>
            <div
              class="sc-1wh81lg-2 dPewCh ql-container ql-snow"
              data-testid="textquill"
            >
              <div
                class="ql-editor ql-blank"
                contenteditable="true"
                data-gramm="false"
                data-placeholder="Set Context: Provide a reason or task or project name so the person knows why you’re requesting feedback"
              >
                <p>
                  <br />
                </p>
              </div>
              <div
                class="ql-clipboard"
                contenteditable="true"
                tabindex="-1"
              />
              <div
                class="ql-tooltip ql-hidden"
              >
                <a
                  class="ql-preview"
                  href="about:blank"
                  rel="noopener noreferrer"
                  target="_blank"
                />
                <input
                  data-formula="e=mc^2"
                  data-link="https://quilljs.com"
                  data-video="Embed URL"
                  type="text"
                />
                <a
                  class="ql-action"
                />
                <a
                  class="ql-remove"
                />
              </div>
            </div>
            <div
              class="sc-1wh81lg-3 dQvbgb"
            >
              0/1000
            </div>
            <span
              class="sc-1ntgl3l-0 fbvXZC"
              textalign="initial"
              textcolor="red"
              textvariant="body2"
            />
          </div>
        </div>
        <div
          class="sc-1sijxax-0 cVvIbq"
        >
          <button
            class="sc-1k2druj-0 eLgfbc"
            data-testid="sendBtn"
            disabled=""
            edges="square"
            type="button"
            variant="solid"
          >
            <span
              class="sc-1ntgl3l-0 eOsHHD"
              textalign="initial"
              textcolor="inherit"
              textvariant="button1"
            >
              Send
            </span>
          </button>
        </div>
      </div>
      <div
        class="sc-tkiih6-9 bFDgcw tips-division"
      >
        <div
          class="sc-uyelpa-6 eorrBQ"
        >
          <div
            class="sc-uyelpa-7 fFEUuk"
          >
            <h3
              class="sc-1ntgl3l-0 lhJlYH sc-tkiih6-22 isRtxH"
              textalign="initial"
              textcolor="inherit"
              textvariant="h3"
              textweight="Strong"
            >
              Tips
            </h3>
            <div
              class="sc-uyelpa-5 evBYe"
            />
          </div>
          <strong
            class="sc-1ntgl3l-0 dDvaST sc-1q8hbsf-1 byiyVE"
            margin="2rem 0 0"
            style="font-size: 14px;"
            textalign="initial"
            textcolor="inherit"
            textweight="Strong"
          >
            Who Can You Request Feedback From?
          </strong>
          <ul
            class="sc-1q8hbsf-2 bzddJq"
            style="margin-top: 0px;"
          >
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                Anyone in your company - your peers, or managers, or cross-functional team members
              </span>
            </li>
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                Clients, vendors, or other stakeholders (via email)
              </span>
            </li>
          </ul>
          <strong
            class="sc-1ntgl3l-0 dDvaST sc-1q8hbsf-1 byiyUX"
            margin="1rem 0 0"
            style="font-size: 14px;"
            textalign="initial"
            textcolor="inherit"
            textweight="Strong"
          >
            What You Can Do with This Feature?
          </strong>
          <ul
            class="sc-1q8hbsf-2 bzddJq"
            style="margin-top: 0px;"
          >
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                <strong>
                  Request for Yourself: 
                </strong>
                Ask others for feedback on your performance or contribution
              </span>
            </li>
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                <strong>
                  Request About Others: 
                </strong>
                Ask someone to give feedback about your teammate
              </span>
            </li>
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                <strong>
                  External Feedback: 
                </strong>
                Enter a client or partner’s email to collect feedback from outside your company
              </span>
            </li>
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                <strong>
                  Set Context: 
                </strong>
                Provide a reason or project name so the person knows why you’re requesting feedback
              </span>
            </li>
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                <strong>
                  Transparent Sharing: 
                </strong>
                If you request feedback about someone else, they will also see the feedback given
              </span>
            </li>
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                <strong>
                  Request at the Right Time: 
                </strong>
                Send requests soon after a task/project ends for more accurate feedback.
              </span>
            </li>
            <li
              class="sc-1q8hbsf-3 cVwZPH"
              style="margin: 0px;"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU"
                style="font-size: 14px;"
                textalign="initial"
                textcolor="inherit"
              >
                <strong>
                  Privacy Note
                </strong>
                : Feedback is visible to the person it’s about. You will also see the feedback if you requested it for yourself or about someone else.
              </span>
            </li>
          </ul>
          <ul
            class="sc-1q8hbsf-2 bzddJq"
            style="padding: 0px;"
          >
            <strong
              class="sc-1ntgl3l-0 dDvaST sc-1q8hbsf-1 byiyUX"
              margin="1rem 0 0"
              style="font-size: 14px;"
              textalign="initial"
              textcolor="inherit"
              textweight="Strong"
            >
              Request Feedback for Yourself
            </strong>
            <span
              class="sc-1ntgl3l-0 hGagSU"
              style="font-size: 14px; margin-top: 0px;"
              textalign="initial"
              textcolor="inherit"
            >
              Context: "Hi Akash, I’d love your feedback on how I handled the stakeholder communication during the Q2 roadmap discussions. Any suggestions for improvement would be appreciated!"
            </span>
            <strong
              class="sc-1ntgl3l-0 dDvaST sc-1q8hbsf-1 byiyUX"
              margin="1rem 0 0"
              style="font-size: 14px;"
              textalign="initial"
              textcolor="inherit"
              textweight="Strong"
            >
              Request Feedback About a Teammate:
            </strong>
            <span
              class="sc-1ntgl3l-0 hGagSU"
              style="font-size: 14px; margin-top: 0px;"
              textalign="initial"
              textcolor="inherit"
            >
              Context: "Hi Rahul, can you share your experience working with Aniket Sharma on the CRM migration project? I’d like to gather multi-perspective feedback for their review cycle."
            </span>
            <strong
              class="sc-1ntgl3l-0 dDvaST sc-1q8hbsf-1 byiyUX"
              margin="1rem 0 0"
              style="font-size: 14px;"
              textalign="initial"
              textcolor="inherit"
              textweight="Strong"
            >
              Request Feedback from a Client:
            </strong>
            <span
              class="sc-1ntgl3l-0 hGagSU"
              style="font-size: 14px; margin-top: 0px;"
              textalign="initial"
              textcolor="inherit"
            >
              Context: "Hi James, we’re collecting feedback to understand how we can improve our collaboration. I’d appreciate it if you could share your thoughts on how Rakesh Pandit has supported you in the recent sprint."
            </span>
          </ul>
        </div>
      </div>
    </div>
  </main>
</div>
`;
